src/main/java/com/o19s/es/ltr/query/LtrQueryBuilder.java

```diff
@@ -114,17 +119,19 @@ protected Query doToQuery(QueryShardContext context) throws IOException {
         if (_features == null || _rankLibScript == null) {
             return new MatchAllDocsQuery();
         }
-        List<String> featureNames = new ArrayList<String>(_features.size());
-        List<Query> asLQueries = new ArrayList<Query>(_features.size());
-        for (QueryBuilder query : _features) {
-            asLQueries.add(query.toQuery(context));
-            featureNames.add(query.queryName());
+        PrebuiltFeature[] features = new PrebuiltFeature[_features.size()];
+        for(int i = 0; i < _features.size(); i++) {
```

Comment:
There are some ways to get around that ([CompletableFutures](http://stackoverflow.com/questions/30117134/aggregate-runtime-exceptions-in-java-8-streams) come to mind), but it is probably not worth it. It might make sense to invest in streams if we can move enough of the transformations there as it would allow us to get rid of a lot of intermediate collections. That being said, this should not be a priority at this point.